docs/wslupath.1
.TH "WSLUPATH" "1" "DATEPLACEHOLDER" "VERSIONPLACEHOLDER" "WSL Utilities"
.SH NAME
wslupath(DEPRECATED) - Component of WSL Utilities
.SH SYNOPSIS
.B wslupath
.RB [ \-dOr ]
.RB [\-D|\-A|\-T|\-S|\-W|\-s|\-su|\-H|\-P|\|\.\|\.\|\.NAME\|\.\|\.\|\.]\fR
.PP
\Bwslupath 
.RB [\-h|\-v|\-R]
.SH DESCRIPTION
This is a WSL Windows path Converter that can convert Windows path to other styles of path\.
.SH COMMANDS
.SS Output Type Options
Without a type, it will detect path style automatically.
.TP
.B -O, --original
print original (Windows) form of path(C:\eWindows\eSystem32)
.TP
.B -d, --doubledash-dir
print Windows form of path with double backslash(C:\e\eWindows\e\eSystem32)
.TP
.B -r, --reg-data
use the registry data to print path[wslpath \-R to know more]
.SS System Path
.B -D, --desktop
print desktop path and exit
.TP
.B -A, --appdata
print AppData path(%APPDATA%) and exit
.TP
.B -T, --temp
print temp path(%TMP%) and exit
.TP
.B -S, --sysdir
print system path and exit
.TP
.B -W, --windir
print Windows path and exit
.TP
.B -s, --start-menu
print Start Menu path and exit
.TP
.B -su, --startup
print Startup path and exit
.TP
.B -H, --home
print user home path (%HOMEPATH%) and exit
.TP
.B -P, --program-files
print Program Files path (%ProgramFiles%) and exit
.SS Other Options
.TP
.B -h, --help
print a simple help
.TP
.B -v, --version
print current version
.TP
.B -R, --avail-reg
print available registry input
.SH AUTHOR
Created by Patrick Wu <me@patrickwu.space>
.SH REPORTING BUGS
Report bugs to <https://github.com/wslutilities/wslu/issues>;
For Ubuntu specific/related bugs, report to <https://bugs.launchpad.net/ubuntu/+source/wslu>.
.PP
.B However, keep in mind this tool is deprecated and bugs reported for this tool will have a high chance to be ignored.
.SH COPYRIGHT
This is free software; you can redistribute it and/or modify it under
the terms of the GNU GPL version 3 or (at your option) any later
version.
There is NO warranty; not even MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
.SH SEE ALSO
wslvar(1), wslu(7)

debian/changelog
wslu (3.2.3-0kali1) kali-dev; urgency=medium

  * wslsys: fix an issue in call_windows_uptime
    (GH: #182)
  * header: fix a distro detection issue in 
    Ubuntu Impish (21.10)
  * debian/control: fix dependency issue in 
    installing wslu (LP: #1926771)
  * docs: include available commands in manpage 
    (GH: #169)
  * project: fix a problem that Powershell output
    non-unicode character (GH: #162)
  * wslview: fix an issue that caused the relative
    Linux path unable to be opened (GH: #165)
  * wslusc: fix an issue that some people unable
    to open their GUI application (GH: #174)
  * header: Avoid error message when the file doesn't
    exist (LP: 1911379)

 -- Patrick Wu <me@patrickwu.space>  Mon, 14 Jun 2021 22:14:49 +0800

wslu (3.2.2-0kali1) kali-dev; urgency=medium

  * docs: include available commands in manpage 
    (GH: #169)
  * project: fix a problem that Powershell output
    non-unicode character (GH: #162)
  * wslview: fix an issue that caused the relative
    Linux path unable to be opened (GH: #165)
  * wslusc: fix an issue that some people unable
    to open their GUI application (GH: #174)

 -- Patrick Wu <me@patrickwu.space>  Mon, 15 Mar 2021 10:52:33 +0800

wslu (3.2.1-0kali1) kali-dev; urgency=medium

  * header: improvements on the system drive detection. (closes #133)
  * tests: update the test system and tests. (closes #125)
  * wslact: add back-compatibility for the auto-mounting feature.
  * wslview: fix an issue when handling http/https
  * wslview: fix a bug that it won't open file:///mnt/c/... or /mnt/c/...
  * wslusc: fix an issue that wsl.exe execute with wrongly passed param (close #138)
  * wslusc: fix the problem that the previous version that GUI shortcut generated won't work.
  * wslsys: fix an issue that the WSL Version shown is wrong.
  * Includes a wslview.desktop (PR #143 and #146)
  * Update the script to fix badly generated shortcut issue (fixes #141)
  * Fix a DPI script related space issue (fixes #144)

 -- Patrick Wu <me@patrickwu.space>  Wed, 23 Dec 2020 22:33:07 +0800

wslu (3.1.1-0kali1) kali-dev; urgency=medium

  * project: add separate Fedora Remix support
  * header: fix `ubuntu.exe` detection issue
  * wslview: mistake in calling a variable (fixes #118 again)
  * wslusc: always use the absolute path (fixes #129)
  * wslusc: fixes a possible white space path issue
  * wslusc: When TMP is empty, try TEMP

 -- Patrick Wu <me@patrickwu.space>  Wed, 01 Jul 2020 22:22:53 +0800

wslu (3.1.0-0kali1) kali-dev; urgency=medium

  * header: use the proper color definition
  * wslfetch: new display for WSL with WSL version displayed
  * wslfetch: use the proper logo for Ubuntu (Reported by @flexiondotorg)
  * wslsys: add WSL version option
  * wslsys: fix an issue in DPI scaling getting wrong value due to the API change in Latest fast ring insider build (Reported by @crramirez)
  * wslact: adding `--mount-options` for auto-mounting feature
  * wslact: fix an i18n related issue in time-sync feature (fixes #113 by @ricardosantos9521)
  * wslgsu: removed temporarily
  * wslgsu: limit the Windows version it can use
  * wslview: properly detect all different protocols and path styles (fixes #118)
  * package: fix alpine dependency issue (fixes #117)

 -- Patrick Wu <me@patrickwu.space>  Wed, 13 May 2020 22:53:03 +0800

wslu (3.0.1-0kali1) kali-dev; urgency=medium

  * Initial release

 -- Patrick Wu <me@patrickwu.space>  Mon, 13 Apr 2020 20:41:40 +0800


src/wslupath.sh
# shellcheck shell=bash
##########   CAUTION   ###########
## wslupath is a legacy cli for backward compatbility.
## Use it unless it is necessary.

style=1
reg_path=0
set_path=""

help_short="wslupath [-dOr] [-D|-A|-T|-S|-W|-s|-su|-H|-P|...NAME...]\nwslupath [-h|-v|-R]"

function path_double_dash {
	new_path="${*//\\/\\\\}"
	echo "$new_path"
}

function general_converter {
	target="$*"

	if [[ $target =~ ^[A-Z]:(\\[^:\\]+)*(\\)?$ ]]; then
		p=$(wslpath -u "${target}")
	elif [[ $target =~ ^$(interop_prefix)[A-Za-z](/[^/]+)*(/)?$ ]]; then
		p=$(wslpath -w "${target}")
	else
		echo "${error} No proper path form detected: ""$*""."
		exit 20
	fi
	echo "$p"
}

function style_path {
	case $style in
		1) p="$(general_converter "$@")";;
		2) p="$*";;
		3) p="$(path_double_dash "$@")";;
	esac
	echo "$p"
}

if [[ $# -eq 0 ]]; then
	echo -e "$help_short"
	exit 20
else
	for args; do
		case $args in
			#styles
			-r|--reg-data) reg_path=1;;
			-O|--original) style=2;;
			-d|--doubledash-dir) style=3;;
			## system location

			-D|--desktop)
			set_path="$(style_path "$(wslvar -l 'Desktop')")"
			break;;
			-A|--appdata)
			set_path="$(style_path "$(wslvar -s APPDATA)")"
			break;;
			-T|--temp)
			set_path="$(style_path "$(wslvar -s TMP)")"
			break;;
			-S|--sysdir)
			set_path="$(style_path "$(wslvar -s windir)"\\System32)"
			break;;
			-W|--windir)
			set_path="$(style_path "$(wslvar -s windir)")"
			break;;
			-s|--start-menu)
			set_path="$(style_path "$(wslvar -l 'Start Menu')")"
			break;;
			-su|--startup)
			set_path="$(style_path "$(wslvar -l 'Startup')")"
			break;;
			-H|--home)
			set_path="$(style_path "$(wslvar HOMEDRIVE)""$(wslvar HOMEPATH)")"
			break;;
			-P|--program-files)
			set_path="$(style_path "$(wslvar -s ProgramFiles)")"
			break;;
			-h|--help) help "$0" "$help_short"; exit;;
			-v|--version) version; exit;;
			-R|--avail-reg) echo "Available registery input:"
			wslvar -L
			exit;;
			*)
			if [[ "$reg_path" == "1" ]]; then
				set_path="$(style_path "$(wslvar -l "$args")")"
			else
				set_path="$(style_path "$args")"
			fi
			break;;
		esac
	done
fi

if [[ "$set_path" == "" ]]; then
	echo "${error}No path input. Aborted."
	exit 21
else
	echo "$set_path"
fi
